### PR TITLE
Add deprecation notice for Android x86 when building for the target

### PR DESCRIPTION
**Pull request by bkonyi**
Related to https://github.com/flutter/flutter/issues/157543

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/gradle.dart [hunk 1]
```diff
@@ -122,6 +122,11 @@ String getAarTaskFor(BuildInfo buildInfo) {
   return _taskFor('assembleAar', buildInfo);
 }
 
+@visibleForTesting
+const String androidX86DeprecationWarning =
+  'Support for Android x86 targets will be removed in the next stable release after 3.27. '
```
Comment (andrewkolos) [inline packages/flutter_tools/lib/src/android/gradle.dart:127]: Should this be `after 3.28`? The 3.28 branch has already been cut, so I we'll be removing support for anything in 3.28.

Comment (bkonyi) [reply packages/flutter_tools/lib/src/android/gradle.dart:127]: The plan is to cherry pick this change so it's shipped with the upcoming stable release. Dart wants to remove x86 support for Dart 3.7.

Comment (andrewkolos) [reply packages/flutter_tools/lib/src/android/gradle.dart:127]: SGTM

**Review by andrewkolos — APPROVED** [review]
LGTM

**Pull request merged**